### ISSUE TITLE
Starting Playlist via MQTT doesn't work if Player is idle

### ISSUE THREAD

**Issue by ghormann** [FalconChristmas/fpp#1029]
v5.x-master-306-g39a4319b (master branch)

* If FFP is currently running a playlist, the the MQTT command "set/playlist/$WHAT/start" will switch the playlist.
* If FPP is idle, the same MQTT command fails to start the playlist 

2021-07-03 19:07:47.607 (51869) [Control] mqtt.cpp:464: Received Mosquitto message: '24580message' on topic '/christmasdev/falcon/player/FPPdev/set/playlist/Test1/start'
2021-07-03 19:07:47.607 (51869) [Control] mqtt.cpp:546: MosquittoClient::CacheSetMessage('/christmasdev/falcon/player/FPPdev/set/playlist/Test1/start', '24580message')
2021-07-03 19:07:47.607 (51869) [Control] mqtt.cpp:288: Publishing message 'Test1' on topic '/christmasdev/falcon/player/FPPdev/playlist/name/status'
2021-07-03 19:07:47.607 (51869) [Control] mqtt.cpp:288: Publishing message '24580' on topic '/christmasdev/falcon/player/FPPdev/playlist/position/status'
2021-07-03 19:07:47.607 (51869) [Control] mqtt.cpp:288: Publishing message '0' on topic '/christmasdev/falcon/player/FPPdev/playlist/repeat/status'
`2021-07-03 19:07:47.615 (51843) [Playlist] playlist/Playlist.cpp:675: Section position 0 is outside of section New`
2021-07-03 19:07:47.620 (51843) [Control] mqtt.cpp:288: Publishing message 'idle' on topic '/christmasdev/falcon/player/FPPdev/status'
2021-07-03 19:07:47.621 (51843) [Control] mqtt.cpp:288: Publishing message '' on topic '/christmasdev/falcon/player/FPPdev/playlist/name/status'
2021-07-03 19:07:47.621 (51843) [Control] mqtt.cpp:288: Publishing message '' on topic '/christmasdev/falcon/player/FPPdev/playlist/section/status'
2021-07-03 19:07:47.621 (51843) [Control] mqtt.cpp:288: Publishing message '0' on topic '/christmasdev/falcon/player/FPPdev/playlist/sectionPosition/status'
2021-07-03 19:07:47.621 (51843) [Control] mqtt.cpp:288: Publishing message '0' on topic '/christmasdev/falcon/player/FPPdev/playlist/repeat/status'

**Comment by MagicMountainMan** [comment]
I just updated to the latest dev version and when I try to start a MQTT playlist it now crashes FPP.

**Comment by ghormann** [comment]
Can you provide the logs and the exact MQTT command used?

**Comment by MagicMountainMan** [comment]
```
2021-07-06 20:51:21.257 (1433) [General] log.cpp:354: =========================================
2021-07-06 20:51:21.261 (1433) [General] log.cpp:355: FPP 5.x-master-317-g791f7e13
2021-07-06 20:51:21.261 (1433) [General] log.cpp:356: Branch: master
2021-07-06 20:51:21.261 (1433) [General] log.cpp:357: =========================================
2021-07-06 20:51:21.264 (1433) [General] fppd.c:503: Creating MQTT
2021-07-06 20:51:21.268 (1433) [Control] mqtt.cpp:268: Using MQTT client id of Falcon Player 
2021-07-06 20:51:21.268 (1433) [Control] mqtt.cpp:292: No CA File specified for MQTT
2021-07-06 20:51:21.269 (1433) [General] fppd.c:515: Creating Scheduler, Playlist, and Sequence
2021-07-06 20:51:21.269 (1433) [General] common.cpp:703: JSON File /home/fpp/media/config/schedule.json does not exist
2021-07-06 20:51:21.269 (1433) [Schedule] Scheduler.cpp:700: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2021-07-06 20:51:21.269 (1433) [Schedule] Scheduler.cpp:701: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2021-07-06 20:51:21.269 (1433) [Schedule] Scheduler.cpp:702: ---- ----------------------- ------------- ------------------- ---------------------------------------------
2021-07-06 20:51:21.277 (1453) [Control] mqtt.cpp:441: HandleConnect() called before can process messages.  Won't Register topics yet.
2021-07-06 20:51:21.283 (1433) [General] fppd.c:518: Creation of Scheduler, Playlist, and Sequence Complete
2021-07-06 20:51:21.426 (1433) [Plugin] src/FPP-HomeAssistant.cpp:22: Initializing Home Assistant Plugin
2021-07-06 20:51:21.431 (1433) [General] common.cpp:703: JSON File /home/fpp/media/config/gpio.json does not exist
2021-07-06 20:51:21.520 (1433) [ChannelOut] channeloutput/channeloutput.c:368: RPIWS281X 0:  Determined range needed 1999 - 3507
2021-07-06 20:51:21.520 (1433) [ChannelOut] channeloutput/channeloutput.c:368: RPIWS281X 0:  Determined range needed 0 - 1733
2021-07-06 20:51:21.521 (1433) [ChannelOut] channeloutput/channeloutput.c:408: Determined range needed 1999 - 3507
2021-07-06 20:51:21.521 (1433) [ChannelOut] channeloutput/channeloutput.c:408: Determined range needed 0 - 1733
2021-07-06 20:51:21.529 (1433) [Command] command.c:73: Initializing Command Module
2021-07-06 20:51:21.532 (1433) [E131Bridge] e131bridge.cpp:313: Universe Count = 0
2021-07-06 20:51:21.632 (1433) [General] fppd.c:659: Checking MQTT
2021-07-06 20:51:21.632 (1433) [Control] mqtt.cpp:412: Mosquitto SetReady()
2021-07-06 20:51:21.632 (1433) [Control] mqtt.cpp:445: Mosquitto Connected.... Will Subscribe to Topics
2021-07-06 20:51:21.632 (1433) [Control] mqtt.cpp:473: MQTT HandleConnect Complete
2021-07-06 20:51:21.632 (1433) [General] fppd.c:667: Starting main processing loop
2021-07-06 20:53:09.774 (1433) [General] fppd.c:596: Shutting down main loop.
2021-07-06 20:53:10.774 (1433) [General] fppd.c:788: Stopping channel output thread.
2021-07-06 20:53:10.774 (1433) [General] fppd.c:792: Main Loop complete, shutting down.
2021-07-06 20:53:12.957 (1433) [General] log.cpp:354: =========================================
2021-07-06 20:53:12.957 (1433) [General] log.cpp:355: FPP 5.x-master-319-g4ff9086b
2021-07-06 20:53:12.957 (1433) [General] log.cpp:356: Branch: master
2021-07-06 20:53:12.957 (1433) [General] log.cpp:357: =========================================
2021-07-06 20:53:12.960 (1433) [General] fppd.c:503: Creating MQTT
2021-07-06 20:53:12.962 (1433) [Control] mqtt.cpp:268: Using MQTT client id of Falcon Player 
2021-07-06 20:53:12.963 (1433) [Control] mqtt.cpp:292: No CA File specified for MQTT
2021-07-06 20:53:12.963 (1433) [General] fppd.c:515: Creating Scheduler, Playlist, and Sequence
2021-07-06 20:53:12.964 (1433) [General] common.cpp:703: JSON File /home/fpp/media/config/schedule.json does not exist
2021-07-06 20:53:12.964 (1433) [Schedule] Scheduler.cpp:700: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2021-07-06 20:53:12.964 (1433) [Schedule] Scheduler.cpp:701: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2021-07-06 20:53:12.964 (1433) [Schedule] Scheduler.cpp:702: ---- ----------------------- ------------- ------------------- ---------------------------------------------
2021-07-06 20:53:12.968 (2750) [Control] mqtt.cpp:441: HandleConnect() called before can process messages.  Won't Register topics yet.
2021-07-06 20:53:12.978 (1433) [General] fppd.c:518: Creation of Scheduler, Playlist, and Sequence Complete
2021-07-06 20:53:13.079 (1433) [Plugin] src/FPP-HomeAssistant.cpp:22: Initializing Home Assistant Plugin
2021-07-06 20:53:13.080 (1433) [General] common.cpp:703: JSON File /home/fpp/media/config/gpio.json does not exist
2021-07-06 20:53:13.151 (1433) [ChannelOut] channeloutput/channeloutput.c:368: RPIWS281X 0:  Determined range needed 1999 - 3507
2021-07-06 20:53:13.151 (1433) [ChannelOut] channeloutput/channeloutput.c:368: RPIWS281X 0:  Determined range needed 0 - 1733
2021-07-06 20:53:13.151 (1433) [ChannelOut] channeloutput/channeloutput.c:408: Determined range needed 1999 - 3507
2021-07-06 20:53:13.151 (1433) [ChannelOut] channeloutput/channeloutput.c:408: Determined range needed 0 - 1733
2021-07-06 20:53:13.154 (1433) [Command] command.c:73: Initializing Command Module
2021-07-06 20:53:13.155 (1433) [E131Bridge] e131bridge.cpp:313: Universe Count = 0
2021-07-06 20:53:13.203 (1433) [General] fppd.c:659: Checking MQTT
2021-07-06 20:53:13.204 (1433) [Control] mqtt.cpp:412: Mosquitto SetReady()
2021-07-06 20:53:13.204 (1433) [Control] mqtt.cpp:445: Mosquitto Connected.... Will Subscribe to Topics
2021-07-06 20:53:13.204 (1433) [Control] mqtt.cpp:473: MQTT HandleConnect Complete
2021-07-06 20:53:13.204 (1433) [General] fppd.c:667: Starting main processing loop
2021-07-06 20:59:50.593 (1401) [General] log.cpp:354: =========================================
2021-07-06 20:59:50.600 (1401) [General] log.cpp:355: FPP 5.x-master-319-g4ff9086b
2021-07-06 20:59:50.600 (1401) [General] log.cpp:356: Branch: master
2021-07-06 20:59:50.600 (1401) [General] log.cpp:357: =========================================
2021-07-06 20:59:50.603 (1401) [General] fppd.c:503: Creating MQTT
2021-07-06 20:59:50.610 (1401) [Control] mqtt.cpp:268: Using MQTT client id of Falcon Player 
2021-07-06 20:59:50.611 (1401) [Control] mqtt.cpp:292: No CA File specified for MQTT
2021-07-06 20:59:50.612 (1401) [General] fppd.c:515: Creating Scheduler, Playlist, and Sequence
2021-07-06 20:59:50.612 (1401) [General] common.cpp:703: JSON File /home/fpp/media/config/schedule.json does not exist
2021-07-06 20:59:50.612 (1401) [Schedule] Scheduler.cpp:700: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2021-07-06 20:59:50.612 (1401) [Schedule] Scheduler.cpp:701: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2021-07-06 20:59:50.612 (1401) [Schedule] Scheduler.cpp:702: ---- ----------------------- ------------- ------------------- ---------------------------------------------
2021-07-06 20:59:50.620 (1441) [Control] mqtt.cpp:441: HandleConnect() called before can process messages.  Won't Register topics yet.
2021-07-06 20:59:50.626 (1401) [General] fppd.c:518: Creation of Scheduler, Playlist, and Sequence Complete
2021-07-06 20:59:50.758 (1401) [Plugin] src/FPP-HomeAssistant.cpp:22: Initializing Home Assistant Plugin
2021-07-06 20:59:50.763 (1401) [General] common.cpp:703: JSON File /home/fpp/media/config/gpio.json does not exist
2021-07-06 20:59:50.841 (1401) [ChannelOut] channeloutput/channeloutput.c:368: RPIWS281X 0:  Determined range needed 1999 - 3507
2021-07-06 20:59:50.841 (1401) [ChannelOut] channeloutput/channeloutput.c:368: RPIWS281X 0:  Determined range needed 0 - 1733
2021-07-06 20:59:50.842 (1401) [ChannelOut] channeloutput/channeloutput.c:408: Determined range needed 1999 - 3507
2021-07-06 20:59:50.842 (1401) [ChannelOut] channeloutput/channeloutput.c:408: Determined range needed 0 - 1733
2021-07-06 20:59:50.850 (1401) [Command] command.c:73: Initializing Command Module
2021-07-06 20:59:50.853 (1401) [E131Bridge] e131bridge.cpp:313: Universe Count = 0
2021-07-06 20:59:50.961 (1401) [General] fppd.c:659: Checking MQTT
2021-07-06 20:59:50.961 (1401) [Control] mqtt.cpp:412: Mosquitto SetReady()
2021-07-06 20:59:50.961 (1401) [Control] mqtt.cpp:445: Mosquitto Connected.... Will Subscribe to Topics
2021-07-06 20:59:50.962 (1401) [Control] mqtt.cpp:473: MQTT HandleConnect Complete
2021-07-06 20:59:50.962 (1401) [General] fppd.c:667: Starting main processing loop
2021-07-06 21:03:15.005 (1482) [General] fppd.c:840: Publishing statistics because of "Startup" = {"status":"OK","uuid":"M1-00000000914e72a7"}
2021-07-07 00:00:00.029 (1401) [General] common.cpp:703: JSON File /home/fpp/media/config/schedule.json does not exist
2021-07-07 00:00:00.029 (1401) [Schedule] Scheduler.cpp:700: Current Schedule: (Status: '+' = Enabled, '-' = Disabled, '!' = Outside Date Range, '*' = Repeat, Stop (G)raceful/(L)oop/(H)ard
2021-07-07 00:00:00.029 (1401) [Schedule] Scheduler.cpp:701: Stat Start & End Dates       Days          Start & End Times   Playlist/Command
2021-07-07 00:00:00.029 (1401) [Schedule] Scheduler.cpp:702: ---- ----------------------- ------------- ------------------- ---------------------------------------------
```
I don't see it trying to start the playlist, but my log level is only on info. Also, it crashes the entire Pi immediately.

The command I'm sending is `FPP/falcon/player/FPP/set/playlist/{playlist name}/start`

**Comment by ghormann** [comment]
Um...  I've reproduced a signal 6 crash but only when not compiled with debugging.....  WIll look into it

**Comment by ghormann** [comment]
I believe this is fixed by 81f804d.   *I've only done limited testing.*

The hint was this complier warning.  Changing "return 0" to "return"  avoid the complier warning and the crash. 
![image](https://user-images.githubusercontent.com/24968726/124850910-7f988d00-df6f-11eb-8734-b601361db979.png)
